front/ui/ui-core/src/components/inputs/ComboBox/ComboBox.tsx

```diff
@@ -142,26 +142,29 @@ const ComboBox = <T,>({
   useOutsideClick(wrapperRef, closeSuggestions);
 
   const inputIcons = useMemo(
-    () => [
-      // Conditionally include the clear icon only when input is not empty
-      ...(value
-        ? [
+    () =>
+      inputProps.readOnly
+        ? []
+        : [
+            // Conditionally include the clear icon only when input is not empty
+            ...(value
+              ? [
+                  {
+                    icon: <XCircle variant="fill" />,
+                    action: clearInput,
+                    className: 'clear-icon',
+                  },
+                ]
+              : []),
+            // chevron of the select
             {
-              icon: <XCircle variant="fill" />,
-              action: clearInput,
-              className: 'clear-icon',
+              icon: <ChevronDown size={small ? 'sm' : 'lg'} />,
+              action: focusInput,
+              className: cx('chevron-icon', {
+                disabled: inputProps.disabled,
+              }),
             },
-          ]
-        : []),
-      // chevron of the select
-      {
-        icon: <ChevronDown size={small ? 'sm' : 'lg'} />,
-        action: focusInput,
-        className: cx('chevron-icon', {
-          disabled: inputProps.disabled,
-        }),
-      },
-    ],
+          ],
     [value]
```

Comment:
Thanks for the PR, ithink it's better to add inputProps in the dependency array and actually most of the variables used in the useMemo 
```suggestion
[inputProps.readOnly, inputProps.disabled, value, small, focusInput]
```

Reply:
fixed

Reply:
We should also add `clearInput` then.

Reply:
done